src/borg/archiver.py

```diff
@@ -2281,6 +2368,23 @@ def build_parser(self, prog=None):
                                help='hex object ID(s) to delete from the repo')
         return parser
 
+    @staticmethod
+    def add_archives_filters_args(subparser):
+        filters_group = subparser.add_argument_group('filters', 'Archive filters can be applied to repository targets.')
+        filters_group.add_argument('-P', '--prefix', dest='prefix', type=prefix_spec, default='',
+                                   help='only consider archive names starting with this prefix')
```

Comment:
if `BORG_REPOSITORY` is set, it would filter all archives with the prefix `repo`, otherwise it would complain about a missing location arg.